the-collab-lab/tcl-54-smart-shopping-list
# Issue 23 | Fix SVG imports through Vite SVGR plugin

## Description

We were pinged by @mtfoley that our SVGs weren't displaying in deployment. We learned that, although it works in localhost, SVG file paths can't be used as an image src and that we had to import them as `ReactComponents` in order to utilize them. In this PR the following are completed:
- All SVGs are imported as React Components and successfully renders in deployment
- Add Item styling with pixels are converted to `rem`
- Item search form includes `e.preventDefault()`

### `vite.config.js`
To import ReactComponent SVGs, `svgr({ exportAsDefault: true })` is replaced with `svgr()`.

I used [FreeCodeCamp](https://www.freecodecamp.org/news/how-to-import-svgs-in-react-and-vite/) on _How to Import SVGs Using the Vite Plugin for SVGR_ and [vite-plugin-svgr](https://www.npmjs.com/package/vite-plugin-svgr) as references.

## Related Issue

Closes #46 

## Acceptance Criteria

- [x]  Solve  the issue on how to import SVGs in React and Vite
- [x] Implement SVGs and formatting into app for deployment

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|   ✓   | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |

## Updates

### Before
![image](https://user-images.githubusercontent.com/89554728/222878280-e3d4c257-562a-41e5-a4a2-7b35797abd3d.png)

### After
![image](https://user-images.githubusercontent.com/89554728/222878289-f35daea2-4765-4a0f-b4f6-9c6979461852.png)

## Testing Steps / QA Criteria

1. In your terminal, run the command `git pull` and then `git checkout ad-svg-import-debug`
2. Then run `npm start`
3. Confirm that deployed app rendering matches localhost:3000 display
